core/core-api/src/main/java/io/fabric8/launcher/core/spi/IdentityProviders.java

```diff
@@ -0,0 +1,57 @@
+package io.fabric8.launcher.core.spi;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.fabric8.launcher.base.identity.Identity;
+import io.fabric8.launcher.base.identity.ImmutableTokenIdentity;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * An utility class for {@link IdentityProviders}.
+ */
+public final class IdentityProviders {
+
+    private IdentityProviders() {
+        throw new IllegalAccessError("utility class");
+    }
+
+    public static Optional<Identity> useDefaultOrElseProvide(final Optional<Identity> defaultIdentity, final IdentityProvider identityProvider, final TokenIdentity authorization, final String provider) {
```

Comment:
I don't think we need this method. It can be easily solved by using 
`defaultIdentity.orElseGet(()->identityProvider.getIdentity(authorization, provider))`

Reply:
indeed!! I forgot to remove it I needed it because getIdentity was throwing checked exceptions (but not anymore) ...